### PR TITLE
Updates and Fixes

### PR DESCRIPTION
**Pull request by Luminarys**
This updates nix to the most recent version (needed for the most recent FreeBSD libc), fixes level triggering on kqueue, and fixes reporting of user events on kqueue.

### CODE REVIEW

File: src/epoll.rs [hunk 1]
```diff
@@ -12,13 +12,15 @@ use std::slice;
...
-use nix::sys::eventfd::{eventfd, EFD_CLOEXEC, EFD_NONBLOCK};
+use nix::sys::epoll::EpollFlags;
+use nix::sys::eventfd::{eventfd, EfdFlags};
 use libc;
 use std::io::{Result, Error, ErrorKind};
 use event::Event;
 use notification::Notification;
 use user_event::UserEvent;
 use channel::{channel, Sender, Receiver};
+use nix_err_to_io_err;
```
Comment (andrewjstone) [inline src/epoll.rs]: It sucks that they removed the conversion from nix::Error to io::Error. Eventually, amy should probably have it's own error type being used here I suppose.

File: src/kqueue.rs [hunk 2]
```diff
@@ -199,47 +196,72 @@ fn event_from_filter(filter: EventFilter) -> Event {
...
+        user_data
+    );
 
     match event {
         Event::Read => vec![ev],
         Event::Write => {
-            ev.filter = EventFilter::EVFILT_WRITE;
-            vec![ev]
+            set_filter(&mut ev, EventFilter::EVFILT_WRITE);
+            vec![KEvent::new(ev.ident(), EventFilter::EVFILT_WRITE, ev.flags(), ev.fflags(), ev.data(), ev.udata())]
```
Comment (andrewjstone) [inline src/kqueue.rs]: Why not just return `vec![ev]` here?

Comment (Luminarys) [reply src/kqueue.rs]: I think you're right, no need to reconstruct the event there.

**Comment by Luminarys** [comment]
wrt to the user event fix, I wasn't able to reliably reproduce it, I just noticed it happening in my own usage on FreeBSD and did some digging. Having more tests around the behavior would definitely be a good idea.

**Comment by Luminarys** [comment]
One thing to note for tests, travis also supports OSX now, so having some explicit tests for kqueue polling might be a good idea.

**Comment by andrewjstone** [comment]
Oh, I didn't realize travis supported that. I did merge in a test for the edge triggering change, but I ran it manually on my macbook. If you have time, a PR to run them on travis would be appreciated, otherwise I'll add them when I get a chance.

**Pull request merged**